### PR TITLE
feat: RPC Starknet call  "get state update"

### PR DESCRIPTION
**Pull request by lambdaclass-user**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?
- Currently, there is no functionality for the RPC call "get_state_update"

Issue Number: #150 
# What is the new behavior?

- Added core functionality for said RPC call.
- Added related CLI command.
- Added related REST API endpoint.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

### CODE REVIEW

File: beerus_cli/tests/cli.rs [hunk 1]
```diff
@@ -2126,6 +2129,97 @@ mod test {
         }
     }
 
+    /// Test the `get_state_update` CLI command.
+    /// Given starknet lightclient returns an error, when query get_state_update, then the error is propagated.
+    /// Error case.
```
Comment (abdelhamidbakhta) [inline beerus_cli/tests/cli.rs:2134]: this is not an error case, please update comment

**Review by LucasLvy — APPROVED** [review]
lgtm

File: beerus_cli/tests/cli.rs (hunk 1 above)
Comment (lambdaclass-user) [reply beerus_cli/tests/cli.rs]: [Fixed](https://github.com/keep-starknet-strange/beerus/pull/176/commits/5e3ca16cc560782e3b62dbfecfddf93ce6ee3848)

**Comment by lambdaclass-user** [comment]
Working on conflicts.

File: beerus_cli/tests/cli.rs (hunk 1 above)
Comment (fkrause98) [reply beerus_cli/tests/cli.rs]: My bad, [this is the commit](https://github.com/keep-starknet-strange/beerus/pull/176/commits/c58756a7d8da8af24e42aeae20da3f1ccb99a139#)

**Comment by fkrause98** [comment]
I'll improve that coverage.

**Pull request merged**